internal/commitremap/commitremap.go

```diff
@@ -0,0 +1,123 @@
+package commitremap
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// Struct to represent a single entry in the commit map
+type CommitMapEntry struct {
+	Old string
+	New string
+}
+
+// Parses the file and returns a map of old commit hashes to new commit hashes
+func ParseCommitMap(filePath string) (*[]CommitMapEntry, error) {
+	commitMap := []CommitMapEntry{}
+
+	// Read the commit-map file
+	content, err := os.ReadFile(filePath)
+	if err != nil {
+		return nil, err
+	}
+	// Split the file content into lines
+	lines := strings.Split(string(content), "\n")
+
+	// Iterate over the lines and parse the old and new commit hashes
+	for _, line := range lines {
+		if strings.TrimSpace(line) == "" {
+			continue
+		}
+
+		fields := strings.Fields(line)
+		if len(fields) != 2 {
+			return nil, fmt.Errorf("invalid line: %s", line)
+		}
+
+		commitMap = append(commitMap, CommitMapEntry{
+			Old: fields[0],
+			New: fields[1],
+		})
+	}
+	return &commitMap, nil
+}
+
+func ProcessFiles(archiveLocation string, prefixes []string, commitMap *[]CommitMapEntry) {
+
+	for _, prefix := range prefixes {
+		// Get a list of all files that match the pattern
+		files, err := filepath.Glob(filepath.Join(archiveLocation, prefix+"_*.json"))
+		if err != nil {
+			log.Fatalf("Error getting files: %v", err)
+		}
+
+		// Process each file
+		for _, file := range files {
+			log.Println("Processing file:", file)
+
+			updateMetadataFile(file, commitMap)
+		}
+	}
+}
+
+func updateMetadataFile(filePath string, commitMap *[]CommitMapEntry) {
```

Comment:
@ssulei7 should we move anything processing local files to it's own internal package ?

Reply:
Probably would be a good idea.

Reply:
@amenocal What would be the advantage of moving to it to an internal package?

Reply:
@kuhlman-labs easier to find things and more organization

Reply:
@kuhlman-labs and adding on to @amenocal 's point, if for whatever reason if we need to add an additional feature or address a bug, it'll be a lot more structured to handle.